pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -583,7 +629,14 @@ func adsPushAll() {
 				}
 			}
 		}
-		//}(client)
+
+		time.AfterFunc(20*time.Second, func() {
+			if pushStatus.End == timeZero && time.Since(pushStatus.Start) > 20*time.Second {
+				out, _ := pushStatus.JSON()
+				adsLog.Warnf("Long push %v %s",
+					time.Since(pushStatus.Start), string(out))
```

Comment:
This is for the case a push takes 20s. You're right, need to set End to avoid repeats.
I've seen few cases where it took a long time. 

Maybe make it 60 sec ? It's a sign pilot is under stress and overloaded, and needs more replicas.

Reply:
I guess tracking push time with a metric would help even more...